src/main/java/org/sunbird/extension/util/TransformJsonUtil.java

```diff
@@ -0,0 +1,698 @@
+    }
+    if (null != enumValues
+        && SunbirdExtensionConstants.OPERATION_MODE_READ.equalsIgnoreCase(operationMode)) {
+      HashBiMap<String, String> biMap = HashBiMap.create(enumValues);
+      enumValues = biMap.inverse();
+    }
+    return enumValues;
+  }
+
+  /**
+   * Method to determine whether to skip transformation for a field Returns true, if there's no
+   * value to transform
+   *
+   * @param fieldValue Object
+   * @return boolean
+   */
+  private static boolean skipTransformationForField(Object fieldValue) {
+    if (null == fieldValue) {
+      return true;
+    }
+    if (fieldValue instanceof String) {
+      if (StringUtils.isBlank((String) fieldValue)) {
+        return true;
+      }
+    }
+    if (fieldValue instanceof List) {
+      if (((List) fieldValue).size() == 0) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private static Object filterValue(
+      String fromField, Map<String, Object> fieldMap, Object fieldValue) {
+    List<Map<String, Object>> filters = null;
+    String filterField = null;
+    Object filteredValue = null;
+    try {
+      filters = (List<Map<String, Object>>) fieldMap.get(TransformationConstants.FILTERS);
+      filterField = (String) fieldMap.get(TransformationConstants.FILTER_FIELD);
+    } catch (Exception e) {
+      throwErrorJsonTransformInvalidFilterConfig(fromField);
+    }
+    if (null == filters || filters.isEmpty() || StringUtils.isBlank(filterField)) {
+      throwErrorJsonTransformInvalidFilterConfig(fromField);
+    }
+
+    List<Map<String, Object>> fromValueList = (List<Map<String, Object>>) fieldValue;
+    if (null == fromValueList || fromValueList.isEmpty()) {
+      throwErrorJsonTransformInvalidInput(fromField);
+    }
+
+    for (Map<String, Object> filter : filters) {
+      fromValueList = applyFilter(fromField, filter, fromValueList);
+    }
+    if (!fromValueList.isEmpty()) {
+      filteredValue = fromValueList.get(0).get(filterField);
+    }
+
+    return filteredValue;
+  }
+
+  private static List<Map<String, Object>> applyFilter(
+      String fromField, Map<String, Object> filter, List<Map<String, Object>> fieldValues) {
+
+    List<Map<String, Object>> filteredList = new ArrayList<Map<String, Object>>();
+
+    String configuredField = (String) filter.get(TransformationConstants.FIELD);
+    List<String> configuredValues = (List<String>) filter.get(TransformationConstants.VALUES);
+    if (null == configuredValues
+        || configuredValues.isEmpty()
+        || StringUtils.isBlank(configuredField)) {
+      throwErrorJsonTransformInvalidFilterConfig(fromField);
+    }
+
+    for (Map<String, Object> fieldValue : fieldValues) {
+      Object value = fieldValue.get(configuredField);
+      if (configuredValues.contains(value)) {
+        filteredList.add(fieldValue);
+      }
+    }
+
+    return filteredList;
+  }
+
+  private static void throwErrorJsonTransformInvalidFilterConfig(String fromField) {
```

Comment:
Use throwServerErrorException in ProjectCommonException.

Reply:
Review comment Incorporated